openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java

```diff
@@ -442,6 +442,170 @@ default Class<Container> getEntityClass() {
     default String getNameColumn() {
       return "fullyQualifiedName";
     }
+
+    @Override
+    default List<String> listBefore(ListFilter filter, int limit, String before) {
+      boolean root = Boolean.parseBoolean(filter.getQueryParam("root"));
+      String condition = filter.getCondition();
+
+      // By default, root will be false. We won't filter the results then
+      if (!root) {
+        return EntityDAO.super.listBefore(filter, limit, before);
+      }
+
+      StringBuilder mysqlCondition = new StringBuilder();
+      StringBuilder psqlCondition = new StringBuilder();
+
+      mysqlCondition.append(String.format("%s ", condition));
+      psqlCondition.append(String.format("%s ", condition));
+
+      mysqlCondition.append("AND er.toId is NULL");
+      psqlCondition.append("AND er.toId is NULL");
```

Comment:
nit: Do we need to do this in this way if they do not differ?

Reply:
done